CMakeLists.txt

```diff
@@ -445,17 +445,22 @@ endif(${WITH_LTTNG})
 option(WITH_OSD_INSTRUMENT_FUNCTIONS OFF)
 
 #option for Babeltrace
-option(HAVE_BABELTRACE "Babeltrace libraries are enabled" ON)
-if(${HAVE_BABELTRACE})
+option(WITH_BABELTRACE "Babeltrace libraries are enabled" ON)
+if(WITH_BABELTRACE)
+  set(HAVE_BABELTRACE ON)
   find_package(babeltrace REQUIRED)
   set(WITH_BABELTRACE ${BABELTRACE_FOUND})
   set(HAVE_BABELTRACE_BABELTRACE_H ${BABELTRACE_FOUND})
   set(HAVE_BABELTRACE_CTF_EVENTS_H ${BABELTRACE_FOUND})
   set(HAVE_BABELTRACE_CTF_ITERATOR_H ${BABELTRACE_FOUND})
-endif(${HAVE_BABELTRACE})
+endif(WITH_BABELTRACE)
+
+option(WITH_LIBZFS "LibZFS is enabled" OFF)
+if(WITH_LIBZFS)
+  set(HAVE_LIBZFS ON)
```

Comment:
@tchaikov 
This was code that was already in the file, and the only thing I did was to convert it also to the std way of setting cmake cli options. 
I'm not using it (yet), because it is unclear to me what functions of ZFS (or btrfs) are really going to be used. And if that makes sense over just using filestore. Especially seeing that we are going to go to BlueStore. But I'll put it on my list look at it.

If you want I can take this out, but rather have all things consistently named.
These are the reference to HAVE_LIBZFS:
```
./src/os/filestore/ZFSFileStoreBackend.h:7:#ifdef HAVE_LIBZFS
./src/os/filestore/FileStore.cc:772:#ifdef HAVE_LIBZFS
./src/os/filestore/ZFSFileStoreBackend.cc:29:#ifdef HAVE_LIBZFS
./src/test/filestore/TestFileStore.cc:56:#ifdef HAVE_LIBZFS
```